### ISSUE TITLE
#[derive(ObjectHash)]

### ISSUE THREAD

**Issue by dtolnay** [cryptosphere/objecthash-rs#15]
[Macros 1.1](https://doc.rust-lang.org/nightly/book/procedural-macros.html) instead of `objecthash_struct!` would make it much easier to hash structs and enums.

**Comment by tarcieri** [comment]
Yes definitely, I've been poking at it. Not entirely trivial though.

**Comment by tarcieri** [comment]
One option here would be to investigate using serde visitors. Then objecthash could work for any type that implements `Serialize`
